framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/ASceneWrapper.java

```diff
@@ -0,0 +1,234 @@
+     */
+    private void removeAnnosFromATypeElement(
+            ATypeElement typeElt, TypeUseLocation loc, AnnotationsInContexts annosToRemove) {
+        String annosToRemoveKey = typeElt.description.toString() + typeElt.tlAnnotationsHere;
+        Set<String> annosToRemoveForLocation = annosToRemove.get(Pair.of(annosToRemoveKey, loc));
+        if (annosToRemoveForLocation != null) {
+            Set<Annotation> annosToRemoveHere = new HashSet<>();
+            for (Annotation anno : typeElt.tlAnnotationsHere) {
+                if (annosToRemoveForLocation.contains(anno.def().toString())) {
+                    annosToRemoveHere.add(anno);
+                }
+            }
+            typeElt.tlAnnotationsHere.removeAll(annosToRemoveHere);
+        }
+
+        // Recursively remove annotations from inner types
+        for (ATypeElement innerType : typeElt.innerTypes.values()) {
+            removeAnnosFromATypeElement(innerType, loc, annosToRemove);
+        }
+    }
+
+    /**
+     * Write the scene wrapped by this object to a file at the given path.
+     *
+     * @param jaifPath the path of the file to be written, but ending in ".jaif". If {@code
+     *     outputformat} is not JAIF, the path will be modified to match.
+     * @param annosToIgnore which annotations should be ignored in which contexts
+     * @param outputFormat the output format to use
+     */
+    public void writeToFile(
+            String jaifPath, AnnotationsInContexts annosToIgnore, OutputFormat outputFormat) {
+        AScene scene = theScene.clone();
+        removeAnnosFromScene(scene, annosToIgnore);
+        scene.prune();
+        String filepath;
+        switch (outputFormat) {
+            case JAIF:
+                filepath = jaifPath;
+                break;
+            case STUB:
+                filepath = jaifPath.replace(".jaif", ".astub");
+                break;
+            default:
+                throw new BugInCF("Unhandled outputFormat " + outputFormat);
+        }
+        new File(filepath).delete();
+        if (!scene.isEmpty()) {
+            // Only write non-empty scenes into files.
+            try {
+                switch (outputFormat) {
+                    case STUB:
+                        SceneToStubWriter.write(this, new FileWriter(filepath));
+                        break;
+                    case JAIF:
+                        IndexFileWriter.write(scene, new FileWriter(filepath));
+                }
+            } catch (IOException e) {
+                throw new UserError("Problem while writing %s: %s", filepath, e.getMessage());
+            } catch (DefException e) {
+                throw new BugInCF(e);
+            }
+        }
+    }
+
+    /**
+     * Obtain the representation of the given class, which can be further operated on to e.g. add
+     * information about a method. This method also updates the metadata stored about the class
```

Comment:
I'm not sure what "metadata" is and how it contrasts with the data stored by this class.  A few methods talk about metadata, but that term is not defined.